lib/widgets/login.dart

```diff
@@ -71,13 +183,28 @@ class _AddAccountPageState extends State<AddAccountPage> {
             constraints: const BoxConstraints(maxWidth: 400),
             child: TextField(
               controller: _controller,
-              onSubmitted: (value) => _onSubmitted(context, value),
+              onSubmitted: (value) => _onSubmitted(context),
               keyboardType: TextInputType.url,
+              autocorrect: false,
+              textInputAction: TextInputAction.go,
+              onEditingComplete: () {
+                // Repeat default implementation by clearing IME compose session…
+                _controller.clearComposing();
+                // …but leave out unfocusing the input in case more editing is needed.
+              },
               decoration: InputDecoration(
                 labelText: 'Your Zulip server URL',
-                suffixIcon: InkWell(
-                  onTap: () => _onSubmitted(context, _controller.text),
-                  child: const Icon(Icons.arrow_forward))))))));
+                hintText: 'your-org.zulipchat.com',
+                errorText: error == null || error.shouldDeferFeedback()
+                  ? null
+                  : error.message(),
+                suffixIcon: IconButton(
+                  onPressed: !_inProgress && (error == null || error.shouldDeferFeedback())
+                    ? () => _onSubmitted(context)
+                    : null, // disabled; errorText will be present
+                  icon: _inProgress
+                    ? const CircularProgressIndicator.adaptive()
+                    : const Icon(Icons.arrow_forward))))))));
```

Comment:
This ends up looking kind of odd, I think. Especially so with the Material progress indicator, because it's somewhat bigger than the arrow icon.

One thing I see looking at the Material guidelines is they don't actually seem to recommend using this suffix/trailing icon for a submit button, as I did when first sketching this screen:
https://m3.material.io/components/text-fields/guidelines#5c8a5f07-b1a5-455f-bf76-7ff0d724f6b0

Perhaps just make it a bit "continue" button below the text field?